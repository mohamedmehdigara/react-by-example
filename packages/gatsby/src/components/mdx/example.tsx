import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
