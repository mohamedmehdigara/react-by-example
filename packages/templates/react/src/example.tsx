import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
