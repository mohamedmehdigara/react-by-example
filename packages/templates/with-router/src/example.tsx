import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
