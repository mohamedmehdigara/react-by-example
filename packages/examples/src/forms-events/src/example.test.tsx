import {
  cleanup,
  fireEvent,
  render,
  RenderResult,
} from '@testing-library/react'
import React from 'react'
import { Events } from './example'

describe('Events', () => {
  let component: RenderResult
  let input: HTMLElement

  beforeEach(() => {
    component = render(<Events />)
    input = component.getByLabelText('try copy/pasting')
  })

  afterEach(() => {
    cleanup()
  })

  it('should snapshot', () => {
    expect(component.container).toMatchSnapshot()
  })

  describe('when focus', () => {
    beforeEach(() => {
      fireEvent.focus(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onFocus')
    })

    describe('when blur', () => {
      beforeEach(() => {
        fireEvent.blur(input)
      })

      it('should display last event', () => {
        expect(component.container).toHaveTextContent('onBlur')
      })
    })
  })

  describe('when blur', () => {
    beforeEach(() => {
      fireEvent.blur(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onBlur')
    })
  })

  describe('when copy', () => {
    beforeEach(() => {
      fireEvent.copy(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onCopy')
    })
  })

  describe('when cut', () => {
    beforeEach(() => {
      fireEvent.cut(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onCut')
    })
  })

  describe('when paste', () => {
    beforeEach(() => {
      fireEvent.paste(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onPaste')
    })
  })

  describe('when keyDown', () => {
    beforeEach(() => {
      fireEvent.keyDown(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onKeyDown')
    })

    describe('when keyUp', () => {
      beforeEach(() => {
        fireEvent.keyUp(input)
      })

      it('should display last event', () => {
        expect(component.container).toHaveTextContent('onKeyUp')
      })
    })
  })

  describe('when keyUp', () => {
    beforeEach(() => {
      fireEvent.keyUp(input)
    })

    it('should display last event', () => {
      expect(component.container).toHaveTextContent('onKeyUp')
    })
  })
})
