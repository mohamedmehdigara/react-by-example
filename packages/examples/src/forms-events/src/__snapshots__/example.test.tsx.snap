// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events should snapshot 1`] = `
<div>
  <div>
    <input
      aria-label="try copy/pasting"
      value="try copy/pasting"
    />
    <pre>
      last event: 
      N/A
    </pre>
  </div>
</div>
`;
